cfme/tests/v2v/test_v2v_migrations.py

```diff
@@ -441,3 +299,283 @@ def test_conversion_host_tags(appliance, v2v_providers):
+                                    host_creds, conversion_tags,
+                                    form_data_vm_obj_single_datastore,
+                                    power_state):
+    # Test VM migration power state and tags are preserved
+    vm = form_data_vm_obj_single_datastore[1][0]
+    if power_state not in vm.mgmt.state:
+        if power_state == 'RUNNING':
+            vm.mgmt.start()
+        elif power_state == 'STOPPED':
+            vm.mgmt.stop()
+    tag = (appliance.collections.categories.instantiate(display_name='Owner *').collections.tags
+        .instantiate(display_name='Production Linux Team'))
+    vm.add_tag(tag)
+    vm.set_retirement_date(offset={'hours': 1})
+
+    infrastructure_mapping_collection = appliance.collections.v2v_mappings
+
+    mapping = infrastructure_mapping_collection.create(form_data_vm_obj_single_datastore[0])
+
+    @request.addfinalizer
+    def _cleanup():
+        infrastructure_mapping_collection.delete(mapping)
+
+    migration_plan_collection = appliance.collections.v2v_plans
+
+    migration_plan = migration_plan_collection.create(
+        name="plan_{}".format(fauxfactory.gen_alphanumeric()), description="desc_{}"
+        .format(fauxfactory.gen_alphanumeric()), infra_map=mapping.name,
+        vm_list=form_data_vm_obj_single_datastore[1], start_migration=True)
+
+    # explicit wait for spinner of in-progress status card
+    view = appliance.browser.create_view(navigator.get_class(migration_plan_collection, 'All').VIEW)
+    wait_for(func=view.progress_card.is_plan_started, func_args=[migration_plan.name],
+        message="migration plan is starting, be patient please", delay=5, num_sec=150,
+        handle_exception=True)
+
+    # wait until plan is in progress
+    wait_for(func=view.get_plan_status, func_args=[migration_plan.name],
+        message="migration plan is in progress, be patient please",
+        delay=5, num_sec=3600)
+
+    view.migr_dropdown.item_select("Completed Plans")
+    view.wait_displayed()
+    logger.info("For plan %s, migration status after completion: %s, total time elapsed: %s",
+        migration_plan.name, view.migration_plans_completed_list.get_vm_count_in_plan(
+            migration_plan.name), view.migration_plans_completed_list.get_clock(
+            migration_plan.name))
+    assert view.migration_plans_completed_list.is_plan_succeeded(migration_plan.name)
+    # check power state on migrated VM
+    rhv_prov = v2v_providers[1]
+    migrated_vm = rhv_prov.mgmt.get_vm((form_data_vm_obj_single_datastore[1][0]).name)
+    assert power_state in migrated_vm.state
+    # check tags
+    vm_obj = appliance.collections.infra_vms.instantiate(migrated_vm.name, rhv_prov)
+    owner_tag = None
+    for t in vm_obj.get_tags():
+        if tag.display_name in t.display_name:
+            owner_tag = t
+    assert owner_tag is not None and tag.display_name in owner_tag.display_name
+    assert 'Never' not in vm_obj.retirement_date
```

Comment:
Maybe a comment here on what ```Never``` means? Is it if a provider hasn't been migrated it would show Never?

Reply:
I'll add a comment on this. `Never` means retirement date not set. If `Never` is not there, that means retirement is set, which is what I want to assert.